traffic_portal/app/src/common/modules/table/serverServerCapabilities/TableServerServerCapabilitiesController.js

```
@@ -0,0 +1,123 @@
+			click: function ($itemScope) {
+				$scope.confirmRemoveCapability($itemScope.sc.serverCapability);
+			}
+		}
+	];
+
+	$scope.addServerCapability = function() {
+		var params = {
+			title: 'Add Server Capability',
+			message: "Please select a capability to add to this server",
+			key: "name"
+		};
+		var modalInstance = $uibModal.open({
+			templateUrl: 'common/modules/dialog/select/dialog.select.tpl.html',
+			controller: 'DialogSelectController',
+			size: 'md',
+			resolve: {
+				params: function () {
+					return params;
+				},
+				collection: function(serverCapabilityService) {
+					return serverCapabilityService.getServerCapabilities();
+				}
+			}
+		});
+		modalInstance.result.then(function(serverCapability) {
+			serverService.addServerCapability(server.id, serverCapability.name)
+				.then(
+					function(result) {
+						messageModel.setMessages(result.data.alerts, false);
+						$scope.refresh(); // refresh the profile parameters table
+					}
+				);
+		}, function () {
+			// do nothing
+		});
+	};
+
+	$scope.confirmRemoveCapability = function(serverCapability, $event) {
+		if ($event) {
+			$event.stopPropagation(); // this kills the click event so it doesn't trigger anything else
+		}
+		var params = {
+			title: 'Remove Capability from Server?',
+			message: 'Are you sure you want to remove the ' + serverCapability + ' capability from this server?'
+		};
+		var modalInstance = $uibModal.open({
+			templateUrl: 'common/modules/dialog/confirm/dialog.confirm.tpl.html',
+			controller: 'DialogConfirmController',
+			size: 'md',
+			resolve: {
+				params: function () {
+					return params;
+				}
+			}
+		});
+		modalInstance.result.then(function() {
+			serverService.removeServerCapability(server.id, serverCapability)
+				.then(
+					function(result) {
+						messageModel.setMessages(result.data.alerts, false);
+						$scope.refresh(); // refresh the profile parameters table
+					}
+				);
+		}, function () {
+			// do nothing
```

Can you log the error message here, so errors aren't swallowed?